CMakeLists.txt

```diff
@@ -6,6 +6,15 @@ endif()
 
 project(userver)
 
+if (USERVER_BUILD_SHARED_LIBS)
```

Comment:
There may be problems with shared linkage of the userver. It  was not tested, some of the system libraries overloads may be skipped due to linkage order.

Either remove the shared build or add shared libraries full testing into CI